test_sysconfig test_parse_makefile_renamed_vars fails when $CFLAGS is set

# Bug report

### Bug description:

In Fedora, when we build and test Python 3.15.0a1, we get:

```
======================================================================
FAIL: test_parse_makefile_renamed_vars (test.test_sysconfig.MakefileTests.test_parse_makefile_renamed_vars)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/python3.15-3.15.0_a1-build/Python-3.15.0a1/Lib/test/test_sysconfig.py", line 824, in test_parse_makefile_renamed_vars
    self.assertEqual(vars, {
    ~~~~~~~~~~~~~~~~^^^^^^^^
        'var1': '-Wall',
        ^^^^^^^^^^^^^^^^
    ...<5 lines>...
        'var3': '',
        ^^^^^^^^^^^
    })
    ^^
AssertionError: {'var1': '-flto=auto -ffat-lto-objects -fexceptions [552 chars]: ''} != {'var1': '-Wall', 'CFLAGS': '-Wall', 'PY_CFLAGS': '-[67 chars]: ''}
Diff is 762 characters long. Set self.maxDiff to None to see it.
```

This is because we set environment variables during the build, like `CFLAGS`.

To reproduce outside of Fedora, you can:

```
[cpython (main %)]$ ./python -m test test_sysconfig
...
== Tests result: SUCCESS ==

[cpython (main %)]$ CFLAGS=-02 ./python -m test test_sysconfig
...
test test_sysconfig failed -- Traceback (most recent call last):
  File ".../cpython/Lib/test/test_sysconfig.py", line 811, in test_parse_makefile_renamed_vars
    self.assertEqual(vars, {
    ~~~~~~~~~~~~~~~~^^^^^^^^
        'var1': '-Wall',
        ^^^^^^^^^^^^^^^^
    ...<5 lines>...
        'var3': '',
        ^^^^^^^^^^^
    })
    ^^
AssertionError: {'var1': -2, 'PY_CFLAGS': '-Wall', 'CFLAGS': '-Wal[64 chars]: ''} != {'var1': '-Wall', 'CFLAGS': '-Wall', 'PY_CFLAGS': [69 chars]: ''}
  {'CFLAGS': '-Wall',
   'LDFLAGS': '-lm',
   'PY_CFLAGS': '-Wall',
   'PY_LDFLAGS': '-lm',
-  'var1': -2,
?           ^

+  'var1': '-Wall',
?          + ^^^^^

   'var2': '-lm',
   'var3': ''}

0:00:00 load avg: 17.34 [1/1/1] test_sysconfig failed (1 failure)

== Tests result: FAILURE ==
```

I believe the tests needs to isolate itself from os.environ.

### CPython versions tested on:

3.15, CPython main branch

### Operating systems tested on:

Linux

<!-- gh-linked-prs -->
### Linked PRs
* gh-140213
<!-- /gh-linked-prs -->

--- a/Lib/test/test_sysconfig.py
+++ b/Lib/test/test_sysconfig.py
@@ -20,7 +20,7 @@ from test.support import (
 )
 from test.support.import_helper import import_module
 from test.support.os_helper import (TESTFN, unlink, skip_unless_symlink,
-                                    change_cwd)
+                                    change_cwd, EnvironmentVarGuard)
 from test.support.venv import VirtualEnvironmentMixin
 
 import sysconfig
@@ -807,7 +807,9 @@ class MakefileTests(unittest.TestCase):
             print("PY_LDFLAGS=-lm", file=makefile)
             print("var2=$(LDFLAGS)", file=makefile)
             print("var3=$(CPPFLAGS)", file=makefile)
-        vars = _parse_makefile(TESTFN)
+        with EnvironmentVarGuard() as env:
+            env.clear()
+            vars = _parse_makefile(TESTFN)
         self.assertEqual(vars, {
             'var1': '-Wall',
             'CFLAGS': '-Wall',

--- a/Misc/NEWS.d/next/Tests/2025-10-16-15-08-58.gh-issue-140210.P9vUP8.rst
+++ b/Misc/NEWS.d/next/Tests/2025-10-16-15-08-58.gh-issue-140210.P9vUP8.rst
@@ -1,0 +1,2 @@
+Make ``test_sysconfig.test_parse_makefile_renamed_vars`` less fragile by
+clearing the environment variables before parsing the Makefile.

